shared/bundle_analysis/parser.py

```diff
@@ -54,66 +54,57 @@ def reset(self):
-                self._parse_event(event)
-
-            # Delete old session/asset/chunk/module with the same bundle name if applicable
-            old_session = (
-                self.db_session.query(Session)
-                .filter(
-                    Session.bundle == self.session.bundle, Session.id != self.session.id
-                )
-                .one_or_none()
-            )
-            if old_session:
-                for model in [Asset, Chunk, Module]:
-                    to_be_deleted = self.db_session.query(model).filter(
-                        model.session == old_session
+        try:
+            self.reset()
+
+            self.session = Session(info={})
+            self.db_session.add(self.session)
+
+            with open(path, "rb") as f:
+                for event in ijson.parse(f):
+                    self._parse_event(event)
+
+                # Delete old session/asset/chunk/module with the same bundle name if applicable
+                old_session = (
+                    self.db_session.query(Session)
+                    .filter(
+                        Session.bundle == self.session.bundle,
+                        Session.id != self.session.id,
                     )
-                    for item in to_be_deleted:
-                        self.db_session.delete(item)
-                self.db_session.delete(old_session)
-                self.db_session.flush()
-
-            # save top level bundle stats info
-            self.session.info = json.dumps(self.info)
-
-            # this happens last so that we could potentially handle any ordering
-            # of top-level keys inside the JSON (i.e. we couldn't associate a chunk
-            # to an asset above if we parse the chunk before the asset)
-            self._create_associations()
-
-        assert self.session.bundle is not None
-        return self.session.id
+                    .one_or_none()
+                )
+                if old_session:
+                    for model in [Asset, Chunk, Module]:
+                        to_be_deleted = self.db_session.query(model).filter(
+                            model.session == old_session
+                        )
+                        for item in to_be_deleted:
+                            self.db_session.delete(item)
+                    self.db_session.delete(old_session)
+                    self.db_session.flush()
+
+                # save top level bundle stats info
+                self.session.info = json.dumps(self.info)
+
+                # this happens last so that we could potentially handle any ordering
+                # of top-level keys inside the JSON (i.e. we couldn't associate a chunk
+                # to an asset above if we parse the chunk before the asset)
+                self._create_associations()
+
+                assert self.session.bundle is not None
+                return self.session.id
+        except Exception as e:
```

Comment:
The diff is just formatting, basically wrapped the entire code block in a `try/except`, inject a plugin name to whatever the exception raised during bundle file parsing is.